src/inference/src/dev/core_impl.cpp

```diff
@@ -377,6 +392,9 @@ ov::Parsed parse_device_config(const std::string& device_name,
             clean_batch_properties(updated_device_name, updated_config, name);
         }
     }
+
+    parsed._config.insert(ov::cache_mode(parsed._core_config.get_cache_mode()));
```

Comment:
There is no need to store in core (is not core property), the mode is stored in compiled model and should be restored from it when read from cache.